### DISCUSSION TITLE
Custom theme break deploy script

### DISCUSSION THREAD

**Discussion by henryavila** [filamentphp/filament#12585] [Help]
### Package

Panel builder

### Package Version

3.2.72

### How can we help you?

Hi, I have this problem.

The custom theme contains:
`@import '../../../vendor/filament/filament/resources/css/theme.css';`
which requires the `composer install` be executed before the frontend build.

But running the `composer install` before the frontend build trigger a error, because vite manifest is not found

```
  Vite manifest not found at: /home/....../public/build/manifest.json
```

Probally someone has alrady faced this problem. How to solve it?

**Comment by leandrocfe** [comment]
`npm i && npm run build`
?

**Reply by henryavila** [reply]
The problem is a cycle dependency.

running `npm run build`  will parse the Filament theme, that requires the file `@import '../../../vendor/filament/filament/resources/css/theme.css'` 

But this file don't exists, because `composer install` has not been executed.

If execute `composer install`  before `npm run build` , it will fail beacuse the web page layout cotains vite command that look for a manifest file created by `npm run build`, but its no execute yeat.

In **local enviroment**, I can run teh comand again after te fail:
- composer install - will fail in dump autoloa, but wil install the files
- npm run build - will not fail because '../../../vendor/filament/filament/resources/css/theme.css'`  now exists
- composer install - will not fail because the manifest files are now generated

But in production deploy thhis error will break the deploy process.

I'm trying to not include the manifest files in git

**Reply by lukas-frey** [reply]
I have a similar issue, when I set a brandLogo in the Service Provider of a Panel, it uses Vite::asset(). And I have the exact same cyclic issue. Anyone found a solution?

**Comment by GustsJ** [comment]
Hi, we fixed the deploys with **almost empty gitted** `manifest.json` with entry and only a file atribute, in build we make build directory and `cp manifest.json` the **gitted** `manifest.json`, run `composer install` and then `npm install `and `npm run build` which overwrites our cp manifest.json, then the custom theme works.

example of manifest.json :
```
{
    "resources/sass/app.scss": {
        "file": "assets/app.css"
     },
   ...more of your resources
}
```
